android/src/main/kotlin/com/mr/flutter/plugin/filepicker/FileUtils.kt

```diff
@@ -0,0 +1,615 @@
+                    }
+                }
+            }
+            if (result == null) {
+                result = uri.path?.substringAfterLast('/')
+            }
+        } catch (ex: Exception) {
+            Log.e(
+                TAG,
+                "Failed to handle file name: $ex"
+            )
+        }
+
+        return result
+    }
+
+    @JvmStatic
+    fun isImage(context: Context, uri: Uri): Boolean {
+        val extension = getFileExtension(context, uri)
+        return (extension != null && (extension.contentEquals("jpg") || extension.contentEquals("jpeg") || extension.contentEquals(
+            "png"
+        ) || extension.contentEquals("WEBP")))
+    }
+
+    private fun getFileExtension(context: Context, uri: Uri): String? {
+        val contentResolver = context.contentResolver
+        val mimeType = contentResolver.getType(uri)
+        return MimeTypeMap.getSingleton().getExtensionFromMimeType(mimeType)
+    }
+
+    private fun getCompressFormat(context: Context, uri: Uri): Bitmap.CompressFormat {
+        val format = getFileExtension(context, uri)
+        return when (format!!.uppercase(Locale.getDefault())) {
+            "PNG" -> Bitmap.CompressFormat.PNG
+            "WEBP" -> Bitmap.CompressFormat.WEBP
+            else -> Bitmap.CompressFormat.JPEG
+        }
+    }
+
+    @JvmStatic
+    fun compressImage(originalImageUri: Uri, compressionQuality: Int, context: Context): Uri {
+        val compressedUri: Uri
+        try {
+            context.contentResolver.openInputStream(originalImageUri).use { imageStream ->
+                val compressedFile = createImageFile(context, originalImageUri)
+                val originalBitmap = BitmapFactory.decodeStream(imageStream)
+                // Compress and save the image
+                val fos = FileOutputStream(compressedFile)
+                originalBitmap.compress(
+                    getCompressFormat(context, originalImageUri),
+                    compressionQuality,
+                    fos
+                )
+                fos.flush()
+                fos.close()
+                compressedUri = Uri.fromFile(compressedFile)
+            }
+        } catch (e: IOException) {
+            throw RuntimeException(e)
+        }
+        return compressedUri
+    }
+
+    @Throws(IOException::class)
+    private fun createImageFile(context: Context, uri: Uri): File {
+        val timeStamp = SimpleDateFormat("yyyyMMdd_HHmmss", Locale.getDefault()).format(Date())
+        val imageFileName = "IMAGE_" + timeStamp + "_"
+        val storageDir = context.cacheDir
+        return File.createTempFile(imageFileName, "." + getFileExtension(context, uri), storageDir)
+    }
+
+    /**
+     * @param uri The Uri to check.
+     * @return Whether the Uri authority is DownloadsProvider.
+     */
+    fun isDownloadsDocument(uri: Uri): Boolean {
+        return "com.android.providers.downloads.documents" == uri.authority
```

Comment:
Would this be more readable?

```suggestion
        return uri.authority == "com.android.providers.downloads.documents"
```